### PR TITLE
feat(auth, medusa): Initial auth module middleware

### PR DESCRIPTION
**Pull request by pKorsholm**
note: This is an initial implementation

### CODE REVIEW

File: packages/auth/src/services/auth-module.ts [hunk 1]
```diff
@@ -118,13 +133,50 @@ export default class AuthModuleService<
...
+        { populate: true }
+      ),
       count,
     ]
   }
 
+  async generateJwtToken(authUserId: string, scope: string): Promise<string> {
+    const authUser = await this.authUserService_.retrieve(authUserId)
+    return jwt.sign({ id: authUser.id, scope }, this.options_.jwt_secret, {
+      expiresIn: "1d",
```
Comment (riqwan) [inline packages/auth/src/services/auth-module.ts]: thought: should this be part of options with a default?

File: packages/medusa/src/utils/authenticate-middleware.ts [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+  ): Promise<void> => {
+    const authTypes = Array.isArray(authType) ? authType : [authType]
+    const authModule = req.scope.resolve<IAuthModuleService>(
+      ModuleRegistrationName.AUTH
+    )
+
+    // @ts-ignore
+    const session: MedusaSession = req.session || {}
+
+    if (authTypes.includes("session")) {
```
Comment (riqwan) [inline packages/medusa/src/utils/authenticate-middleware.ts]: suggestion: constantize "session" and "bearer"?

File: packages/medusa/src/utils/authenticate-middleware.ts [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+      if (session.auth && session.auth[authScope]) {
+        const authUser = await authModule.retrieveAuthUser(
+          session.auth[authScope].user_id
+        )
```
Comment (srindom) [inline packages/medusa/src/utils/authenticate-middleware.ts]: thought: if this fails it should probably result in a 401 and not a 404 so maybe worth handling that error

File: packages/medusa/src/utils/authenticate-middleware.ts [hunk 4]
```diff
@@ -0,0 +1,79 @@
...
+        const matches = authHeader.match(re)
+
+        if (matches) {
+          const tokenType = matches[1]
+          const token = matches[2]
+          if (tokenType.toLowerCase() === "bearer") {
+            const authUser = await authModule.retrieveAuthUserFromJwtToken(
+              token,
+              authScope
+            )
```
Comment (srindom) [inline packages/medusa/src/utils/authenticate-middleware.ts]: same as above

**Review by srindom — APPROVED** [review]
LGTM

**Review by riqwan — APPROVED** [review]
lgtm!

**Pull request merged**